tests/ui/type-inference/regression-issue-81317.rs

```diff
@@ -0,0 +1,40 @@
+// Regression test for #81317: type can no longer be infered as of 1.49
```

Comment:
Document the reason this test is failing so someone reading or modifying it in the future doesn't have to reestablish context from scratch

Reply:
How about now?

Reply:
thanks, that's great. Just need to pacify tidy line length checks now